Removing index_dev.php

lenonleite opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 5.x)
* a.b for any bug fixes (e.g. 4.4, 5.1)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [X]
| Deprecations?                          | [X]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [X] <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/mautic-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation-new/pull/126
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

#### Description:

This PR removes the `index_dev.php` and edits environment variables to detect if the environment is `dev`, `prod`, or `testing` and render the assets accordingly. also it define if `debugging` should be enabled or not.


<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->

#### Steps to test this PR:

<!--
This part is really important. If you want your PR to be merged, take the time to write very clear, annotated and step by step test instructions. Do not assume any previous knowledge - testers may not be developers.
-->
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. You should see in gitpod that we are not using `index_dev.php` but `index.php`
3. You can see debugging features if you are in git pod


<!--
If you have any deprecations, list them here along with the new alternative.
If you have any backwards compatibility breaks, list them here.
-->

lenonleite committed:
Removing index_dev.php

lenonleite committed:
Add new config file to environment
refactor in index.php to work in developer and production just changing parameters in environment.php

lenonleite committed:
remove lines that use replace to index_dev.php in php and js files.
Comments line that needs some more effort than remove line

lenonleite committed:
remove lines that use replace to index_dev.php in php and js files.
Comments line that needs some more effort than remove line

lenonleite committed:
change comment of lines gitpod.yml

lenonleite committed:
edit gitpod to run in develop environment

lenonleite committed:
fix edit parameters in gitpod pepiline

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

fedys commented on app/bundles/CoreBundle/Loader/EnvironmentHandler.php:
I would follow the Symfony standards instead of developing our own. Let's use the approach mentioned in the Symfony doc. See https://symfony.com/doc/4.4/configuration.html#selecting-the-active-environment

lenonleite replied:
@fedys , hello,

I was checking the problem in the test and came up with some doubts. Now test not work because need more variables to work like, DB_HOST, DB_NAME, etc etc...

You said that .env would have just APP_ENV=prod and APP_DEBUG=0, but the system needs more basic variables to work, and the example that I saw just in bootstrap.php read .env and not .env.test or .env.local. But I have three approaches to fix it,

  1. Create a .env with all essential variables for work
  2. Use .env.test or .env.local with all essential variables and override with variables of .env.
  3. I will have three env files .env.prod, .env.dev, .env.test, and the official .env will be just to set what kind of env I need

fedys replied:
Ideally we should end up with the approach that is described in the Symfony documentation. Within this PR I would do it the following way if possible:

`.env`:  contains defaults for all environments.
```ini
APP_ENV=prod
APP_DEBUG=0
```

`.env.test`:  contains defaults for test environment. We need to make sure that this file is read by the test suite. I can see it is using just `.env` or `.env.dist` files now. See https://github.com/mautic/mautic/blob/5.x/app/bundles/CoreBundle/Test/EnvLoader.php#L17
```ini
DB_HOST=127.0.0.1
DB_PORT=3306
DB_NAME=mautictest
DB_USER=root
DB_PASSWD=
MAUTIC_DB_PREFIX=test_
MAUTIC_ENV=test
MAUTIC_ADMIN_USERNAME=admin
MAUTIC_ADMIN_PASSWORD=mautic
```

`.env.test.local`:  not in Git. Contains customized values for test environment. We need to make sure that this file is read by the test suite.
```ini
DB_HOST=some-hostname
...
```

`.env.local`:  not in Git. Contains customized values.
```ini
APP_ENV=dev
APP_DEBUG=1
```

fedys commented on index.php:
```diff
@@ -8,12 +8,14 @@
 use Symfony\Component\HttpFoundation\Request;
 
-ErrorHandler::register('prod');
+$config = (new EnvironmentHandler())->getEnvParameters();
```
```suggestion
require dirname(__DIR__).'/config/bootstrap.php';
```
Let's load the environmental variables the Symfony way. See https://github.com/symfony/recipes/blob/main/symfony/framework-bundle/4.4/public/index.php#L7 and https://github.com/symfony/recipes/blob/main/symfony/framework-bundle/4.4/config/bootstrap.php

lenonleite replied:
Hey @fedys, thank you so much for reviewing, I'm reading and trying to understand, and it's clear. 

I have a question, the item 1, when you say similar, are you talking about parts of code with environment right?

fedys replied:
Correct 👍

mabumusa1 replied:
@fedys is it a good idea to include the lines that [handle proxies](https://github.com/symfony/recipes/blob/41d2cf13ea897a08b0f376d20011a69a73ec0366/symfony/framework-bundle/4.4/public/index.php#L15) into this PR or make a new PR for them?

fedys replied:
I would stick with changes connected with the environmental variables in this PR. Let's create a separate PR for proxies related code.

fedys commented on index.php:
```diff
@@ -8,12 +8,14 @@
+ErrorHandler::register($config['ENV']);
 
-$kernel   = (new MiddlewareBuilder(new AppKernel('prod', false)))->resolve();
+$kernel   = (new MiddlewareBuilder(new AppKernel($config['ENV'], $config['DEBUG'])))->resolve();
```
```suggestion
$kernel   = (new MiddlewareBuilder(new AppKernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG'])))->resolve();
```
See https://github.com/symfony/recipes/blob/main/symfony/framework-bundle/4.4/public/index.php#L23

fedys reviewed (changes requested):
Thanks @lenonleite for a great improvement!
Let's improve it, so it is closer to Symfony standards. 

1. Change the `bin/console` file so it looks similar to https://github.com/symfony/recipes/blob/main/symfony/console/4.4/bin/console
2. Let's follow https://symfony.com/doc/4.4/configuration.html#selecting-the-active-environment when it comes to `.env`, `.env.local` and `.env.local` files.
    - `.env` will be renamed to `.env.test`
    - create a new `.env` file that will currently contain just `APP_ENV=prod` and `APP_DEBUG=0` and will no longer be ignored by Git.
    - `.env.local` will be ignored by Git allowing us to override `APP_ENV` and `APP_DEBUG`.
3. See my code comments.

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

refactor index
add boostrap.php
copy index to scaffold

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

refactor index
add boostrap.php
copy index to scaffold

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

refactor index
add boostrap.php
copy index to scaffold

remove .env from gitgnore and ad .env.test

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

refactor index
add boostrap.php
copy index to scaffold

remove .env from gitgnore and ad .env.test
change getenv() to $_SERVER to get right value
change name of env.dist to correct name
and removing environment.php

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

refactor index
add boostrap.php
copy index to scaffold

remove .env from gitgnore and ad .env.test
change getenv() to $_SERVER to get right value
change name of env.dist to correct name
and removing environment.php
remove class enviroment helper

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

refactor index
add boostrap.php
copy index to scaffold

remove .env from gitgnore and ad .env.test
change getenv() to $_SERVER to get right value
change name of env.dist to correct name
and removing environment.php
remove class enviroment helper

lenonleite committed:
edit js compiled
removed index_dev.php reference from composer and add some lines in mautic-setup.sh to change configurations of environment file

fix mautic-setup.sh

remove part url with index_dev.php

remove index_dev.php from scafford folder

remove dev comments
remove reference in htaccess root and scaffold folder
copy index.php from root to scaffold folder

It was created  parameter environment handler in Core Bundle

Remove dump and add comments

fix reference array type

coping index.php in root folder to index.php in scaffold folder

refactor index
add boostrap.php
copy index to scaffold

remove .env from gitgnore and ad .env.test
change getenv() to $_SERVER to get right value
change name of env.dist to correct name
and removing environment.php
remove class enviroment helper

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value MAUTIC_TABLE_PREFIX

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

remove env.dist

copy bin console to scaffold

copy index.php from root folder to scaffold

fedys commented:
@lenonleite thanks a lot for addressing my suggestions! The changes look good. 👍  Please resolve the conflicts with 5.x branch, so the CI pipeline is triggered. Once all the GitHub checks are green, I will test the PR.

lenonleite committed:
Merge remote-tracking branch 'origin/5.x' into index_dev-removed

# Conflicts:
#	app/bundles/CoreBundle/Test/EnvLoader.php

lenonleite commented:
@fedys and @mabumusa1 I did merge. and thank you for your support. =)

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

remove env.dist

copy bin console to scaffold

copy index.php from root folder to scaffold

fixing tests

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

remove env.dist

copy bin console to scaffold

copy index.php from root folder to scaffold

fixing tests

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

remove env.dist

copy bin console to scaffold

copy index.php from root folder to scaffold

fixing tests

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

remove env.dist

copy bin console to scaffold

copy index.php from root folder to scaffold

fixing tests

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

remove env.dist

copy bin console to scaffold

copy index.php from root folder to scaffold

fixing tests

php fixcs

lenonleite committed:
default env value with APP_ENV=prod

remove env.dist

writing real values to env.test based on ddev enviroment

remove part of the script not necessary in mautic-setup

include env value

remove env.dist

copy bin console to scaffold

copy index.php from root folder to scaffold

fixing tests

php fixcs, include config folder to composer fixcs check

mabumusa1 committed:
Merge branch '5.x' into index_dev-removed

mabumusa1 reviewed (dismissed):
Changes looks good, Thanks @lenonleite

lenonleite committed:
come back with configuration line

lenonleite committed:
Merge remote-tracking branch 'origin/index_dev-removed' into index_dev-removed

lenonleite committed:
remove config line

fedys commented on app/bundles/CoreBundle/Test/EnvLoader.php:
```diff
@@ -14,7 +14,7 @@ final class EnvLoader
+        $envFile = file_exists($root.'.env.test.local') ? $root.'.env.test.local' : $root.'.env.test';
         $dotenv  = new Dotenv();
         $dotenv->load($envFile);
     }
```
Let's replace the method like:
```php
    /**
     * Loads the env variables from .env(.*) files for PHPUNIT tests.
     */
    public static function load(): void
    {
        (new Dotenv())->loadEnv(__DIR__.'/../../../../.env');
    }
```
It will allow you to specify **only** those environmental variables in `.env.test.local` that **need to be overridden**.

We need to add the following line:
```xml
    <server name="APP_ENV" value="test" force="true" />
```
to https://github.com/mautic/mautic/blob/5.x/app/phpunit.xml.dist#L55, so it loads **test environment**.

fedys commented on .env.dist:
This file can be removed as it was replaced by `.env.test` file.

fedys reviewed (changes requested):
Thanks a ton @lenonleite! I tested your PR in my local environment and it worked as expected. There is a small issue when it comes to the test environment. Please see my comments.

fedys commented on .env:
```diff
@@ -0,0 +1,3 @@
+# .env
```
```suggestion
# In all environments, the following files are loaded if they exist,
# the latter taking precedence over the former:
#
#  * .env                contains default values for the environment variables needed by the app
#  * .env.local          uncommitted file with local overrides
#  * .env.$APP_ENV       committed environment-specific defaults
#  * .env.$APP_ENV.local uncommitted environment-specific overrides
#
# Real environment variables win over .env files.
#
# DO NOT DEFINE PRODUCTION SECRETS IN THIS FILE NOR IN ANY OTHER COMMITTED FILES.
#
# Run "composer dump-env prod" to compile .env files for production use (requires symfony/flex >=1.2).
# https://symfony.com/doc/current/best_practices.html#use-environment-variables-for-infrastructure-configuration

```
Let's add the comment to ease the understanding the way Dotenv works.

fedys commented on .gitignore:
```diff
@@ -12,10 +12,10 @@
 !.ddev/commands/web/dd
 .php_cs.cache
-/.env
+/.env.local
+/.env.*.local
```
```suggestion
/.env.local
/.env.local.php
/.env.*.local
```

fedys commented on .php-cs-fixer.php:
```diff
@@ -9,6 +9,7 @@
     ->in(__DIR__.'/app/middlewares')
     ->in(__DIR__.'/app/migrations')
     ->in(__DIR__.'/plugins')
+    ->in(__DIR__.'/config')
```
```suggestion
```
Let's move the `config/bootstrap.php` to `app/config/bootstrap.php`

fedys commented on app/assets/scaffold/files/example.gitignore:
```diff
@@ -12,10 +12,10 @@
 !.ddev/commands/web/dd
 .php_cs.cache
-/.env
+/.env.local
+/.env.*.local
```
```suggestion
/.env.local
/.env.local.php
/.env.*.local
```

fedys commented on app/bundles/CoreBundle/Test/MauticMysqlTestCase.php:
```diff
@@ -376,7 +376,12 @@ private function wasRollbackSuccessful(): bool
     private function getTablePrefix(): string
     {
-        return self::$container->getParameter('mautic.db_table_prefix');
+        $prefix = self::$container->getParameter('mautic.db_table_prefix');
+        if (is_null($prefix)) {
+            return '';
+        }
+
+        return $prefix;
```
```suggestion
return (string) self::$container->getParameter('mautic.db_table_prefix');
```
Let's simplify it.

fedys commented on app/composer.json:
We will need to remove `[project-root]/.env.dist` here and add other .env* files if needed.

fedys commented on config/bootstrap.php:
Let's move the `config/bootstrap.php` to `app/config/bootstrap.php`

lenonleite committed:
remove unnecessary env, editing gitgnore to it.
change bootstrap to app/config
editing bin console tu news changes.
fix index.php to work with bootstrap in different folder.
change file responsible to get env test.
edit phpunit file with new env data

fedys committed:
Revert .env.test removal

fedys committed:
.env.test.local for DDEV

lenonleite committed:
remove unnecessary env, editing gitgnore to it.
change bootstrap to app/config
editing bin console tu news changes.
fix index.php to work with bootstrap in different folder.
change file responsible to get env test.
edit phpunit file with new env data
edit mautic-setup.sh to copy new env test to root folder
edit env.test.local

fedys committed:
PHPDoc updated

fedys committed:
Copy .env to assets/scaffold/files/.env

fedys committed:
Copy .env to assets/scaffold/files/.env

fedys reviewed (dismissed):
Thanks @lenonleite! I don't see any issues when it comes to the way the `.env.*` files are merged together now. In my local environment I tested changing both `APP_ENV` and `APP_DEBUG` via adding `.env.local` and it worked like a charm for both HTTP and CLI envs. Also adding `.env.test.local` worked as expected. I was able to specify only the variables that I needed to override. Great job 👍

fedys commented:
@escopecz @mollux I'm done with my code review and tests in my local environment (I'm not using DDEV). Please take a look if there are any issues.

escopecz commented:
The code changes look great! Please drop a note about this change in https://github.com/mautic/mautic/blob/5.x/UPGRADE-5.0.md.

It will also need some docs updates. See for example https://developer.mautic.org/#environments

@RCheesley shall the change be done in the old docs, new docs or both?

RCheesley commented:
> It will also need some docs updates. See for example https://developer.mautic.org/#environments
> 
> @RCheesley shall the change be done in the old docs, new docs or both?

This should be made in the new docs please! https://github.com/mautic/developer-documentation-new for PRs (you can use Gitpod to review them) and https://mautic-developer.readthedocs.io/ for the front-end view of the default branch!

lenonleite commented:
I'm working on documentation, and I will send a new PR soon

lenonleite committed:
documentation

lenonleite committed:
Merge remote-tracking branch 'origin/index_dev-removed' into index_dev-removed

lenonleite committed:
improve documentation

Add documentation to repository developer-documentation-new with Pull Request
https://github.com/mautic/developer-documentation-new/pull/126 - https://mautic-developer.readthedocs.io/en/latest/

lenonleite commented:
@RCheesley and @escopecz  I added documentation here and has a new Pr to https://github.com/mautic/developer-documentation-new/
I don't know if is perfect, but tell me what I need to do if necessary

=)

RCheesley commented:
I'll take a look at it this afternoon, thank you @lenonleite 🚀

escopecz commented:
I had a hickup when testing this. I use the table prefix on my instance and I have this in `app/config/local.php`:
```php
'db_table_prefix' => 'mautic_',
```
When I checked out this PR I could not log in with an error that the db.user table does not exist. It did not take the table prefix into account. When I added this into `.env.local` then the page loaded:
```ini
MAUTIC_TABLE_PREFIX=mautic_
```

Once I did that I can change the `APP_ENV` and `APP_DEBUG` variables and the application behaved as expected. With `APP_ENV=dev` I was able to see the Symfony debug toolbox bar in the bottom and JS dependencies were loaded one by one. With `APP_ENV=prod` the debug bar was gone, JS dependencies were squashed into app.js and libraries.js (plus some JS plugins loaded asynchronously) and the page load was much faster.

Can we do something to fix the `MAUTIC_TABLE_PREFIX` issue?

lenonleite commented:
@escopecz thank you for valid it. 
I will check it today. =)

lenonleite committed:
Work in:

https://mautic.atlassian.net/browse/TPROD-454
https://mautic.atlassian.net/browse/TPROD-449
https://mautic.atlassian.net/browse/TPROD-454

Changing deprecated methods like
exec
executeUpdate
execute

Validate with empty method

lenonleite committed:
Merge remote-tracking branch 'origin/5.x' into index_dev-removed

lenonleite commented:
@escopecz I fixed the problem and merged with the last 5-x and new errors started to appear.  
I think that it isn't about my fix, because tested it before merge and everything was ok, but I will keep looking at it.

mabumusa1 commented:
> @escopecz I fixed the problem and merged with the last 5-x and new errors started to appear. I think that it isn't about my fix, because tested it before merge and everything was ok, but I will keep looking at it.

@lenonleite there is an issue with the tests since Friday, I do not think they are related to this PR

mabumusa1 committed:
Merge branch '5.x' into index_dev-removed

escopecz reviewed (approved):
Works well for me now. Thank you! 👍

fedys reviewed (approved):
👍

mabumusa1 committed:
Merge branch '5.x' into index_dev-removed
